### PR TITLE
Update Hydroquebec for multi contracts accounts

### PR DESCRIPTION
**Pull request by titilambert**
**Description:**
Update Hydroquebec for multi contracts accounts

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#6392

### CODE REVIEW

File: source/_components/sensor.hydroquebec.markdown [hunk 1]
```diff
@@ -52,3 +59,15 @@ Configuration variables:
   - **yesterday_total_consumption**: Yesterday total consumption
   - **yesterday_lower_price_consumption**: Yesterday lower price consumption
   - **yesterday_higher_price_consumption**: Yesterday higher price consumption
+
+<p class='note'>
```
Comment (Landrash) [inline source/_components/sensor.hydroquebec.markdown]: There's no real reason for this to be in a note since it most of the text on the page.

**Comment by titilambert** [comment]
Fixed !

**Comment by Landrash** [comment]
Thank you :)

**Comment by g146m026** [comment]
I don't know if is due to modification but now only the period_length monitored variables works. All other throw an exception.

there the traceback
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/usr/local/lib/python3.5/dist-packages/homeassistant/helpers/entity_component.py", line 359, in async_process_entity
    new_entity, self, update_before_add=update_before_add
  File "/usr/local/lib/python3.5/dist-packages/homeassistant/helpers/entity_component.py", line 189, in async_add_entity
    yield from self.hass.loop.run_in_executor(None, entity.update)
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.5/dist-packages/homeassistant/components/sensor/hydroquebec.py", line 152, in update
    self._state = round(self.hydroquebec_data.data[self.type], 2)
KeyError: 'period_length'

thanks you for all the work you're doing!

**Comment by titilambert** [comment]
@g146m026  Did you add your contract number in the HA configuration ?

**Comment by g146m026** [comment]
Hello thank you for your anwser.
I enter my contract number and i got these error for all my variable setnin
the configuration file.
But i can see randomly one information in the dashboard...

I installed your pyhydriquebec with pip and all seem to work.

Im from quebec city so if you talj French it will be more easy for me.

Best regard
Carl


Le 12 mars 2017 10:03 PM, "Thibault Cohen" <notifications@github.com> a
écrit :

**Comment by djfjeff** [comment]
I get the same here, I added the contract number and I now get the following in the log for multiple keys :

`
17-03-13 07:09:40 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 233, in _step
    result = coro.throw(exc)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/helpers/entity_component.py", line 359, in async_process_entity
    new_entity, self, update_before_add=update_before_add
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/helpers/entity_component.py", line 189, in async_add_entity
    yield from self.hass.loop.run_in_executor(None, entity.update)
  File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/components/sensor/hydroquebec.py", line 152, in update
    self._state = round(self.hydroquebec_data.data[self.type], 2)
KeyError: 'period_total_bill'
`

**Comment by titilambert** [comment]
@djfjeff I found the issue !

**Comment by titilambert** [comment]
@djfjeff PR created ;)

**Comment by djfjeff** [comment]
@titilambert I don't know if I did something wrong when I added your patch here but now I get this on restart :

> 17-03-13 08:14:20 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up platform hydroquebec
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/helpers/entity_component.py", line 153, in _async_setup_platform
    entity_platform.schedule_add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/components/sensor/hydroquebec.py", line 110, in setup_platform
    sensors.append(HydroQuebecSensor(hydroquebec_data, variable, name))
  File "/srv/homeassistant/lib/python3.4/site-packages/homeassistant/components/sensor/hydroquebec.py", line 129, in __init__
    self._update()
AttributeError: 'HydroQuebecSensor' object has no attribute '_update'

Here is the change that I made :

> diff hydroquebec.py /home/homeassistant/hydroquebec.py 
129,130d128
<         self._update()

**Comment by djfjeff** [comment]
@titilambert Sorry, ignore my previous comment, typo on my side, everything works fine, thanks!

**Comment by titilambert** [comment]
@djfjeff I'm not sure that is working. I don't get any update :/

**Comment by djfjeff** [comment]
@titilambert To be honest, I didn't check if update were made correctly at regular interval, I mostly use this sensor as a "check 1 or 2 times a month".

I was more confirming that the sensor was starting correctly and not throwing error in the log.

**Comment by titilambert** [comment]
@djfjeff I just checked and I got updates. So I can confirm the patch

**Comment by MartinHjelmare** [comment]
Please don't discuss in merged or closed PRs. Open an issue.

**Pull request merged**